host/entry.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2018 The ZikiChomgo Authors. All rights reserved.  Use of this source
+// code is governed by a license that can be found in the License file.
+
+package host
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"zikichombo.org/sio/libsio"
+	"zikichombo.org/sound"
+	"zikichombo.org/sound/ops"
+	"zikichombo.org/sound/sample"
+)
+
+// Entry is the type of a connection to host sound I/O capabilities.
+//
+// Entries may either be high level (eg AudioFlinger) or low level (eg ALSA hw).
+//
+// Multiple entries may exist for a given host.  see Names() in the relevant
+// entry_{runtime.GOOS}.go file for details.
+type Entry interface {
+	// Name returns the name of the entry and should be a valid
+	// name for the host.
+	Name() string
+	// DefaultForm returns the default form.  In case one form is not
+	// implemented or uniform accross all supported functionality (input, output, device),
+	// output default should be returned.
+	DefaultForm() sound.Form
+
+	// DefaultSampleCodec returns the default sample codec.  In case one sample codec
+	// is not implemented or uniform accross all supported functionality, output default
+	// should be returned.
+	DefaultSampleCodec() sample.Codec
+
+	// DefaultBufSize returns the default size of the buffer provided to
+	// the caller for interactions.
+	DefaultBufSize() int
+
+	// CanOpenSource returns true if OpenSource does not return
+	// ErrUnspported.
+	CanOpenSource() bool
+
+	// OpenSource starts capturing audio.
```

Comment:
Perhaps (but probably not) rename `OpenSource` to `Capture` or `Record`, or `OpenInputDev`? And do the same with `CanOpenSource`.

Reply:
I think it's better to consider this after a merge, so people have a reasonable place to work with what's there.  We didn't seem to disagree anyway, marking resolved.